src/background/browser-adapters/runtime-adapter.ts

```diff
@@ -0,0 +1,6 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+export type RuntimeAdapter = {
```

Comment:
Currently `getRuntimeId` seems to only be used from `DetailsViewController` and `getRuntimeLastError` seems to only be called from `TabController` - this seems to be contrary to interface segregation.

Reply:
Done

Reply:
Should I break `RuntimeAdapter` into two types, one function each? My original approach for this type was on the semantic of what we are using from chrome API more than who's using it. I do see your point though, I'm just not sure I like the idea to break this into one-function-type

Reply:
I suppose that if we were to follow ISP strictly, we'd essentially have a `DetailsViewControllerBrowserAdapter` interface that contains everything that `DetailsViewController` consumes (that isn't otherwise in a common interface that DVC fully consumes), and a `TabControllerBrowserAdapter` that contains everything that `TabController` consumes (that isn't otherwise in a common interface that TC fully consumes)... not sure if we need/want to get to that point in a single refactoring.

Reply:
I agree I would prefer to do this on a separated PR as the current `RuntimeAdapter` is already small than the original `ChromeAdapter`, i.e. the classes that you mention "depend" on less things that they don't use, or from another point of view, this is already better than what we had before